diff -Nur sg3_utils-1.37_old/src/sg_logs.c sg3_utils-1.37/src/sg_logs.c
--- sg3_utils-1.37_old/src/sg_logs.c	2018-07-16 20:25:42.111255897 +0800
+++ sg3_utils-1.37/src/sg_logs.c	2018-07-16 20:28:08.235249900 +0800
@@ -169,7 +169,8 @@
            "    --transport|-T    decode transport (protocol specific port "
            "0x18) log page\n"
            "    --verbose|-v    increase verbosity\n"
-           "    -N|--new    use new interface\n"
+           "    --new|-N        use new interface\n"
+           "    --old|-O        use old interface (use as first option)\n"
            "    --version|-V    output version string then exit\n\n"
            "Performs a SCSI LOG SENSE (or LOG SELECT) command and decodes "
            "the response.\n");
